example/src/main/java/org/folio/tlib/example/Storage.java

```diff
@@ -0,0 +1,116 @@
+package org.folio.tlib.example;
+
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.validation.RequestParameter;
+import io.vertx.ext.web.validation.RequestParameters;
+import io.vertx.ext.web.validation.ValidationHandler;
+import io.vertx.sqlclient.RowIterator;
+import io.vertx.sqlclient.Tuple;
+import io.vertx.sqlclient.templates.SqlTemplate;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.UUID;
+import org.folio.tlib.postgres.PgCqlField;
+import org.folio.tlib.postgres.PgCqlQuery;
+import org.folio.tlib.postgres.TenantPgPool;
+
+public class Storage {
+
+  TenantPgPool pool;
+
+  Storage(Vertx vertx, String tenant) {
+    pool = TenantPgPool.pool(vertx, tenant);
+  }
+
+  private String getMyTable(TenantPgPool pool) {
+    return pool.getSchema() + ".mytable";
+  }
+
+  Future<Void> init(JsonObject tenantAttributes) {
+    if (!tenantAttributes.containsKey("module_to")) {
+      return Future.succeededFuture(); // doing nothing for disable
+    }
+    Future<Void> future = pool.query(
+            "CREATE TABLE IF NOT EXISTS " + getMyTable(pool)
+                + "(id UUID PRIMARY KEY, title TEXT, index_title TEXT)")
+        .execute().mapEmpty();
+    JsonArray parameters = tenantAttributes.getJsonArray("parameters");
+    if (parameters != null) {
+      for (int i = 0; i < parameters.size(); i++) {
+        JsonObject parameter = parameters.getJsonObject(i);
+        if ("loadSample".equals(parameter.getString("key"))
+            && "true".equals(parameter.getString("value"))) {
+          future = future.compose(x ->
+              pool.preparedQuery("INSERT INTO " + getMyTable(pool)
+                      + "(id, title, index_title) VALUES ($1, $2, $3)")
+                  .execute(Tuple.of(UUID.randomUUID(), "First title", "first title")).mapEmpty()
+          );
+          future = future.compose(x ->
+              pool.preparedQuery("INSERT INTO " + getMyTable(pool)
+                      + "(id, title, index_title) VALUES ($1, $2, $3)")
+                  .execute(Tuple.of(UUID.randomUUID(), "Second title", "second title")).mapEmpty()
+          );
+        }
+      }
+    }
+    return future;
+  }
+
+
+  Future<Book> getBook(UUID id) {
+    return SqlTemplate.forQuery(pool.getPool(), "SELECT * FROM " + getMyTable(pool)
```

Comment:
@adamdickmeiss Can we avoid string concat when using templates and use parameters mapping (.mapFrom) instead for things like table and schema names? If so, it would be cool to pull out templates to a PSFS text blocks to make them more readable.